### ISSUE TITLE
Returning false in after_* callback won't cancel later callbacks

### ISSUE THREAD

**Issue by kgorin** [rails/rails#20422]
The [documentation](http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html) says
>If a before_* callback returns false, all the later callbacks and the associated action are cancelled. **If an after_* callback returns false, all the later callbacks are cancelled.**

but in fact even if you return false in after_* callback none of the later callbacks will be cancelled. 

Here is gist to check that out against master https://gist.github.com/kgorin/467ac79f8abf6f4d50bf, the behaviour is same in 3.2.21 and 4.2.1, so it seems like it's been like that for a long time.

That's happening because class methods in `After` class don't consider callback return values in similar way `Before` class does and thus check for halted chain always returns false.

I've looked through discussion in PR about `throw :abort` https://github.com/rails/rails/pull/17227, but that doesn't seem to work for after_* callbacks. Also it's not clear what should happen if after_* callback returns false or throws, should it be rolled back or just later callbacks cancelled. Probably it's good to be able to roll back if one of the after_ callbacks fail.

So is it a bug or a documentation problem?

**Comment by claudiob** [comment]
Hello @kgorin ! If you take a look at the [edge documentation](http://edgeapi.rubyonrails.org/classes/ActiveRecord/Callbacks.html) (which mirrors the current master) you will see that the paragraph has been replaced by:

> If a before_* callback throws :abort, all the later callbacks and the associated action are cancelled.

The sentence you highlighted is gone so… in principle… all is correct.

However… you are perfectly right in sensing something weird. While working on #17227, @dhh already pointed it out [in a comment](https://github.com/rails/rails/pull/17227#issuecomment-59449385):

> Ultimately, I think you should be able to halt the after_* chain as well. But we don’t need to deal with it in the same PR.

In short, #17227 only dealt with `before_` callbacks. If you'd like to open a new PR and suggest changes to `after_` (and `around_`?) callbacks, go for it! :ribbon:

**Comment by rafaelfranca** [comment]
Closing since this is a feature request and we don't get feature requests on the issue tracker. But feel free to implement it.

**Comment by kgorin** [comment]
Hi @claudiob
I see 2 points here:
1. Shouldn't that documentation piece be backported to 4.2 branch so that it was less misleading?
2. I see some questions on how implementation of after_* callback chain halting should look like, I guess I'll ask them on mailing list then.

**Comment by claudiob** [comment]
@kgorin Maybe I misunderstood your comment. #17227 only affects Rails >= 5.0.

Are you saying that even in Rails 4.2, the following statement is false?

> If an after_* callback returns false, all the later callbacks are cancelled.

**Comment by kgorin** [comment]
@claudiob yes, it is indeed false. I am actually coming from all the way back to 3.2.21 where I found this problem. Just double checked on 4.2.1 https://github.com/rails/rails/commit/f1ccb2e6ecb5486179ee6b20438d562ac45de4f4 — works the same.

**Comment by rafaelfranca** [comment]
Yes. Documentation were wrong. We need to backport it.

**Comment by rafaelfranca** [comment]
@kgorin could you send a PR?

**Comment by kgorin** [comment]
@rafaelfranca sure
